test/fixtures/rules/std_rule_cases/L004.yml

```diff
@@ -80,6 +80,18 @@ indented_comments:
     FROM spam
 
 indented_comments_default_config:
+  # TODO partial fix - this requires some more attention
+  # This is because of TODO fix indents after text:
+  # https://github.com/sqlfluff/sqlfluff/pull/590#issuecomment-739484190
+  # Lint fails on the tabs that come before the comments (expects spaces)
+  #  Errors Found: [
+  #    <SQLLintError: rule L004 pos:(2, 7), #fixes: 0, description:
+  #      Incorrect indentation type found in file. The indent occurs after other text,
+  #      so a manual fix is needed.>,
+  #    <SQLLintError: rule L004 pos:(3, 15), #fixes: 0,
+  #      description: Incorrect indentation type found in file. The indent occurs after
+  #      other text, so a manual fix is needed.>]
+  # Not visible to bare eye but here tab indent is replaced with spaces
```

Comment:
I had a look at this one. This seems to be a case that is consciously left for the user to fix by hand. It would be nice to be able to assert exactly this: that it applies some fixes but then lint fails with that particular error. This is exactly the type of case where ignoring with line level noqa won't cut it because the wrong kind of indent in the beginning of the line is expected to be fixed, but in the middle of the same line not.

Same for the other case in this file, just swapping space <-> tab.

Reply:
I think in general it's definitely a good idea to start asserting that fix_str also always passes lint.

But this test would fail if that is enforced

I think the actual options are:

1. test these cases separately:
   - test with only fail_str with only unfixable content
   - test with fail_str & fix_str with only fixable content
2. extend the test framework to support testing both things in the same test case

With option 1 I feel like we're losing something because we can't test complicated situations where a bug would surface only when both fixable & unfixable sections are present in the source query.